### PR TITLE
Fix redundant list alloc in LotteryNumbers

### PR DESCRIPTION
**Pull request by andrievsky**
Fix redundant list alloc in LotteryNumbers

### CODE REVIEW

File: hexagonal/src/main/java/com/iluwatar/hexagonal/domain/LotteryNumbers.java [hunk 1]
```diff
@@ -85,11 +84,10 @@ public static LotteryNumbers create(Set<Integer> givenNumbers) {
    * @return numbers as comma separated string
    */
   public String getNumbersAsString() {
-    List<Integer> list = new ArrayList<>();
-    list.addAll(numbers);
     StringBuilder builder = new StringBuilder();
+    Iterator<Integer> iterator = numbers.iterator();
     for (int i = 0; i < NUM_NUMBERS; i++) {
-      builder.append(list.get(i));
+      builder.append(iterator.next());
```
Comment (IAmPramod) [inline hexagonal/src/main/java/com/iluwatar/hexagonal/domain/LotteryNumbers.java]: Why we don't have a check for `iterator.hasNext()`

Comment (andrievsky) [reply hexagonal/src/main/java/com/iluwatar/hexagonal/domain/LotteryNumbers.java]: The same reason we didn't check a list boundaries.

File: hexagonal/src/main/java/com/iluwatar/hexagonal/domain/LotteryNumbers.java [hunk 2]
```diff
@@ -85,11 +84,10 @@ public static LotteryNumbers create(Set<Integer> givenNumbers) {
    * @return numbers as comma separated string
    */
   public String getNumbersAsString() {
-    List<Integer> list = new ArrayList<>();
-    list.addAll(numbers);
     StringBuilder builder = new StringBuilder();
+    Iterator<Integer> iterator = numbers.iterator();
```
Comment (npathai) [inline hexagonal/src/main/java/com/iluwatar/hexagonal/domain/LotteryNumbers.java]: Can't we directly use foreach loop and not depend on `NUM_NUMBERS`?

Comment (iluwatar) [reply hexagonal/src/main/java/com/iluwatar/hexagonal/domain/LotteryNumbers.java]: I think so too. However, I'll accept this PR as an improvement and a step into the right direction.

**Comment by iluwatar** [comment]
Thanks @andrievsky :+1:

**Pull request merged**